docker/README.md

```diff
@@ -30,62 +28,80 @@ See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
 
+### Run Firefly
 
-## HOW-TO
+    cd cm/firefly
+    docker-compose up firefly
 
-**Monitor Firefly logs**
+Firefly is now up and running at http://localhost:8080/firefly/
 
-    docker-compose logs -f firefly
 
+### Run Firefly in DevMode
 
-**Interactive bash shell in the container**
+> :info: **If you are using Mac Docker Desktop**:
+> Docker on Mac has had some performance issues since the beginning. These are related to volume performance, 
+> the way volumes are mounted, and the underlying osxfs filesystem.  
+> Release 4.6 fixes this with the introduction of virtiofs.  
+> 
+> To use virtiofs, you must have macOS 12.2 or later and the new Virtualization Framework enabled.
+> Go to Docker Desktop -> Preferences -> Experimental Features 
+> and enable the two features.
 
-    docker-compose exec firefly bash
+DevMode is Firefly development mode.  When running, it will detect changes in the source code, and automatically
+deploy it.  Reload the page in your browser to see the changes.  
 
+    cd cm/firefly
+    docker-compose up dev
 
-**Cleanly shutdown Firefly**
+DevMode only applies to client-side code.  If you made changes to server-side(.java) code, you need to rerun `docker-compose up dev` 
+to have it rebuild and redeploy.
 
-    docker-compose down
+#### Debugger
 
-**Build Firefly from source**
+Client side debugging is done via the browser's DevTool.  
 
-    docker-compose build firefly
+To debug server side code, attach your debugger to a remote JVM on localhost:5050.  Of course, this only works
+while DevMode is running.
 
-If you've made changes to the source, this will pick up the changes and create a new Docker image with your changes.  
-Because Firefly uses multi-stage builds, Docker is smart enough to reuse unaffected layers to speed up the build process.  
-However, if you want a complete clean build, add `--no-cache` to the above command.  
-For more details on docker-compose usage, go here: https://docs.docker.com/compose/reference/ 
 
-BUILD_TAG variable defaults to 'latest'.  You can tag this build by setting it to something else.
+### HOW-TO
 
+**Interactive bash shell in the container**
 
-**Build Firefly from source with arguments**
+    docker-compose exec firefly bash
 
-By default, Firefly builds includes documentation and help pages.  To build only Firefly, 
 
-    docker-compose build --build-arg target=war firefly
+**Cleanly shutdown Firefly**
+
+    docker-compose down
 
-`target` is the gradle task used when building.  There are others, like `test`.
+This will stop Firefly and removes everything created by `up`.
 
 
 **To publish this build to DockerHub**
 
-    docker-compose push
+    docker-compose push firefly
 
 **To run Firefly from DockerHub**
 
 If you want to run a version available from DockerHub.  
 Set BUILD_TAG to the version you want, then 
 
-    docker-compose pull && docker-compose up -d
+    export BUILD_TAG=latest
```

Comment:
is there a way to put `BUILD_TAG` on the command line?

Reply:
Sure, I can suggest a one-line approach.